src/test/fuzz/fuzz.h

```diff
@@ -20,28 +20,49 @@
 
 using FuzzBufferType = Span<const uint8_t>;
 
-using TypeTestOneInput = std::function<void(FuzzBufferType)>;
+using TypeTestOneInput = std::function<bool(FuzzBufferType)>; /* returns true for useful test */
 using TypeInitialize = std::function<void()>;
 using TypeHidden = bool;
 
 void FuzzFrameworkRegisterTarget(std::string_view name, TypeTestOneInput target, TypeInitialize init, TypeHidden hidden);
 
 inline void FuzzFrameworkEmptyInitFun() {}
 
+/** Fuzz target without initialization function that always succeeds. */
 #define FUZZ_TARGET(name) \
     FUZZ_TARGET_INIT(name, FuzzFrameworkEmptyInitFun)
 
+/** Fuzz target without initialization function that returns bool (false = uninteresting test). */
+#define FUZZ_PARTIAL_TARGET(name) \
```

Comment:
Maybe worth noting that this will only work for libFuzzer? (or i guess any engine that uses the libFuzzer harness and respects the -1 return value)